core/src/test/scala/org/apache/spark/scheduler/CoarseGrainedSchedulerBackendSuite.scala

```diff
@@ -120,6 +124,54 @@ class CoarseGrainedSchedulerBackendSuite extends SparkFunSuite with LocalSparkCo
     }
   }
 
+  // Here we just have test for one happy case instead of all cases: other cases are covered in
+  // FsHistoryProviderSuite.
+  test("custom log url for Spark UI is applied") {
+    val conf = new SparkConf()
+      .set(CPUS_PER_TASK, 2)
+      .set(UI.CUSTOM_EXECUTOR_LOG_URL, getCustomExecutorLogUrl(includeFileName = true))
+      .setMaster("local-cluster[4, 3, 1024]")
```

Comment:
You don't need to start 4 executors for this test, right?

I was going to suggest mocking the stuff outside of `CoarseGrainedSchedulerBackend` but that looks a bit complicated. Kinda sucks to need a full SparkContext here.

Reply:
Yeah I guess initializing SparkContext is required for now. Will check whether we can reduce the number of executors here. Thanks!

Reply:
The number doesn't seem to affect the test - I reduce the count to 0 and test still passed. I guess executor will not launch since we don't run any job here.

Reply:
But why are the executors not launching? There's nothing I can see in the configuration that indicates that. 

Are you sure they're actually not launching, instead of just not launching fast enough given what the rest of the test is doing?

I really would prefer something explicit that says "executors will not launch" instead of this.

Reply:
Ah OK I see what you are saying. I guess change the count to 0 will launch no worker and no executor will be launched (as there's no worker). Would we be OK with this?

Reply:
If it works with a count of 0, sounds good.